src/main/java/com/lorepo/icplayer/client/PrintableContentParser.java

```diff
@@ -129,21 +157,67 @@ public String generatePrintableHTMLForPage(Page page, boolean randomizeModules,
 		if (randomizeModules) {
 			randomizePrintables(pagePrintables);
 		}
-		
-		String result = "<div class=\"printable_page\">";
+		String result = "";
 		for (IPrintableModuleModel printable: pagePrintables) {
 			result += printable.getPrintableHTML(showAnswers);
 		}
+		return result;
+	}
+	
+	private String wrapPrintablePage(String content, int pageWidth, int pageHeight) {
+		String pageDivCss = "width:" 
+				+ Integer.toString(pageWidth) 
+				+ "px; height:" 
+				+ Integer.toString(pageHeight) 
+				+ "px;";
+		
+		content = applyHeaderAndFooter(content, true);
+		
+		String result = "<div class=\"printable_page\" style=\"" + pageDivCss + "\">";
+		result += content;
 		result += "</div>";
 		return result;
 	}
-	
-	public String generatePrintableHTML(Content contentModel, boolean randomizePages, boolean randomizeModules, boolean showAnswers) {
-		List<Page> pages = contentModel.getPages().getAllPages();
-		String result = generatePrintableHTMLForPages(pages, randomizePages, randomizeModules, showAnswers);
+	
+	private String applyHeaderAndFooter(String content, boolean fullHeight) {
+		String result = fullHeight ? "<table style='width:100%; height:100%'>" : "<table>";

```

Comment:
Właściwie to wygląda jak idealne miejsce na wykorzystanie:
http://www.gwtproject.org/javadoc/latest/com/google/gwt/safehtml/client/SafeHtmlTemplates.html